13 накинуть абстракций

churchhellah opened the pull request:
Добавлен класс Stack.java, который используется в качестве обоймы для оружия - в Stack помещаются объекты класса Ammo (патроны).
Добавлен enum Ammo, который содержит свойства патрона.
Добавлен enum ShootType, содержащий типы стрельбы - одиночный выстрел или выстрел очередью.
Добавлен абстрактный класс AbstractWeapon для работы с оружием.
Добавлен Singleton Weapons с фабричным методом для создания оружия.
Добавлен интерфейс Warrior.
Добавлен абстрактный класс AbstractWarrior.
Добавлены классы-наследники AbstractWarrior: Soldier, Captain и General.
В классе Team добавлена логика добавления членов команды по вероятности с помощью Random.
В классе App удалено создание оружия, членов команд и самих команд

churchhellah committed:
Создаю Generic Stack

churchhellah committed:
Создаю Generic Stack

Commit:
накинуть абстракций #13

Commit:
накинуть абстракций #13

Commit:
накинуть абстракций #13

churchhellah committed:
Создаю Generic Stack

Commit:
накинуть абстракций #13

churchhellah committed:
Создаю Generic Stack

churchhellah committed:
Создаю Generic Stack

churchhellah committed:
issue #13

churchhellah committed:
13-накинуть-абстракций
Обновил диаграмму классов в README.md

churchhellah committed:
подключил логгер
в pom.xml добавил зависимости log4j и slf4j
в Battle.java добавил логирование
в Team.java перенес аннотацию @Getter для возможности получать список членов команды

churchhellah committed:
в классе AbstractWarrior.java в методе attack логика перезарядки работала так, что оружие всегда перезаряжалось только одним типом патронов. Заменил эту логику вызовом метода reload (метод абстрактный и каждый наследник сам реализует логику перезарядки).
Сделал так, что каждый тип войны перезаряжает оружие своим типом патронов.

churchhellah committed:
в классе AbstractWarrior.java добавил логгер
чтобы выводить информацию о наносимом уроне и критическом уроне оппонентам

showgazer commented on pom.xml:
```diff
@@ -29,5 +29,10 @@
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-collections4</artifactId>
+        <version>4.5.0-M2</version>
```
давай вынесем версии артефактов в переменные (блок <properties>)

churchhellah replied:
сделал ([коммит](https://github.com/churchhellah/battle/pull/14/commits/6594d1a0dee5d24461664fdcb4dd9196df5d1829))

showgazer commented on src/main/java/org/battle/models/Team.java:
```diff
@@ -0,0 +1,47 @@
+
+    // Метод получения живого участника из members
+    public Warrior getAliveMember() {
+        for (Warrior member : members) {
```
давай здесь попробуем fluent методы
members.
stream
filter
or else

churchhellah replied:
сделал ([коммит](c6fd3c525c5162b50b92b16be33a27fe5a3f2227))

showgazer commented on src/main/java/org/battle/services/Battle.java:
```diff
@@ -1,8 +1,8 @@
-package org.battle;
+package org.battle.services;
```
тесты на все это дело не свалились?

churchhellah replied:
тесты пишу [в отдельной ветке](https://github.com/churchhellah/battle/tree/17-Написать-Unit-тесты)
пока худо-бедно работают, но я их еще хочу дополнить

showgazer commented on src/main/java/org/battle/services/Battle.java:
```diff
@@ -13,23 +13,23 @@ public void start() {
+            if (character2.isKilled()) {
                 // TODO подключить логгер
-                System.out.println(character2.getName() + " повержен!");
+                System.out.println(character2.getClass()+": character2 повержен!");
```
логгер)

churchhellah replied:
надеюсь, тут про то, что логгер надо подключить...
если да, то это [в отдельной ветке](https://github.com/churchhellah/battle/tree/14-подключить-логгер-к-новой-версии-программы)
или опять я поперемешивал коммиты?... 🥸

showgazer commented on src/main/java/org/battle/services/Battle.java:
```diff
@@ -13,23 +13,23 @@ public void start() {
-                if (!character1.isAlive()) {
+                if (character1.isKilled()) {
                     // TODO подключить логгер
-                    System.out.println(character1.getName() + " повержен!");
```
логгер)

churchhellah replied:
надеюсь, тут про то, что логгер надо подключить...
если да, то это [в отдельной ветке](https://github.com/churchhellah/battle/tree/14-подключить-логгер-к-новой-версии-программы)
или опять я поперемешивал коммиты?... 🥸

churchhellah committed:
добавил тесты

churchhellah committed:
pom.xml:
вынес версии артефактов в переменные (блок properties)

churchhellah committed:
Team.java:
в методе getAliveMember заменил конструкцию for {if...else} на fluent конструкцию .stream().filter(Warrior -> !Warrior.isKilled()).findAny().orElse(null)

churchhellah committed:
Merge 13-накинуть-абстракций into 14-подключить-логгер-к-новой-версии-программы

churchhellah committed:
pom.xml:
вынес версии артефактов, помеченных "logging" в переменные (блок properties)

churchhellah committed:
Merge pull request #19 from churchhellah/14-подключить-логгер-к-новой-версии-программы

14 подключить логгер к новой версии программы

churchhellah committed:
Merge 13-накинуть-абстракций into 17-Написать-Unit-тесты

churchhellah committed:
pom.xml:
вынес версии артефактов junit в переменные (блок properties)

churchhellah committed:
Merge pull request #20 from churchhellah/17-Написать-Unit-тесты

17 написать unit тесты
